### PR TITLE
feat: improve zooming

### PR DESCRIPTION
**Pull request by ooystein**
Improve zoom speed on mobile, add zoom center point for scroll zooming and double click zooming.
Refactor viewer.service to use ModeService more consistently.

### CODE REVIEW

File: src/lib/src/core/viewer-service/viewer.service.ts [hunk 1]
```diff
@@ -314,86 +326,122 @@ export class ViewerService implements OnInit {
   /**
    * Switches to DASHBOARD-mode, repositions pages and removes max-width on viewer
    */
-  toggleToDashboard(): void {
+  private toggleToDashboard(): void {
```
Comment (raymondkarstensen) [inline src/lib/src/core/viewer-service/viewer.service.ts]: toggleToDashboard can't be private because it is used by ViewerComponent unit test.

Comment (ooystein) [reply src/lib/src/core/viewer-service/viewer.service.ts]: I have fixed the test now.

File: src/lib/src/viewer/viewer.component.spec.ts [hunk 2]
```diff
@@ -216,37 +216,36 @@ describe('ViewerComponent', function () {
...
-
+  it('should change to PAGE-mode when single-click in DASHBOARD-mode', fakeAsync(() => {
+    modeService.mode = ViewerMode.DASHBOARD;
     const firstOverlay = viewerService.getOverlays()[0];
     const clickEvent = createClickEvent(firstOverlay);
     clickService.click(clickEvent);
     tick(1000);
-    expect(comp.mode).toBe(ViewerMode.PAGE);
+    expect(modeService.mode).toBe(ViewerMode.PAGE);
   }));
```
Comment (raymondkarstensen) [inline src/lib/src/viewer/viewer.component.spec.ts]: Should this test be moved to ModeService?

File: src/lib/src/viewer/viewer.component.spec.ts [hunk 3]
```diff
@@ -216,37 +216,36 @@ describe('ViewerComponent', function () {
    * Singleclicks
    **************************************/
 
-  it('should change to PAGE-mode when singleclicking in DASHBOARD-mode', fakeAsync(() => {
-    viewerService.toggleToDashboard();
-
+  it('should change to PAGE-mode when single-click in DASHBOARD-mode', fakeAsync(() => {
```
Comment (raymondkarstensen) [inline src/lib/src/viewer/viewer.component.spec.ts]: Should this test be moved to ModeService?

File: src/lib/src/viewer/viewer.component.spec.ts [hunk 4]
```diff
@@ -216,37 +216,36 @@ describe('ViewerComponent', function () {
...
+    modeService.mode = ViewerMode.DASHBOARD;
     const firstOverlay = viewerService.getOverlays()[0];
     const clickEvent = createClickEvent(firstOverlay);
     clickService.click(clickEvent);
     tick(1000);
-    expect(comp.mode).toBe(ViewerMode.PAGE);
+    expect(modeService.mode).toBe(ViewerMode.PAGE);
   }));
 
   it('should change to dashboard-mode when single-click in page-mode', fakeAsync(() => {
```
Comment (raymondkarstensen) [inline src/lib/src/viewer/viewer.component.spec.ts]: Should this test be moved to ModeService?

File: src/lib/src/viewer/viewer.component.spec.ts [hunk 5]
```diff
@@ -216,37 +216,36 @@ describe('ViewerComponent', function () {
...
     const firstOverlay = viewerService.getOverlays()[0];
     const clickEvent = createClickEvent(firstOverlay);
     clickService.click(clickEvent);
     tick(1000);
-    expect(comp.mode).toBe(ViewerMode.DASHBOARD);
+    expect(modeService.mode).toBe(ViewerMode.DASHBOARD);
   }));
 
 
   it('should change to dashboard-mode when single-click in zoomed-in page-mode', fakeAsync(() => {
```
Comment (raymondkarstensen) [inline src/lib/src/viewer/viewer.component.spec.ts]: Should this test be moved to ModeService?

File: src/lib/src/viewer/viewer.component.spec.ts [hunk 6]
```diff
@@ -256,42 +255,42 @@ describe('ViewerComponent', function () {
    **************************************/
 
   it('should change to PAGE-mode when doubleclicking in DASHBOARD-mode', fakeAsync(() => {
```
Comment (raymondkarstensen) [inline src/lib/src/viewer/viewer.component.spec.ts]: Should this test be moved to ModeService?

File: src/lib/src/viewer/viewer.component.spec.ts [hunk 7]
```diff
@@ -256,42 +255,42 @@ describe('ViewerComponent', function () {
...
     const firstOverlay = viewerService.getOverlays()[0];
     const clickEvent = createClickEvent(firstOverlay);
     clickService.click(clickEvent);
     clickService.click(clickEvent);
     tick(1000);
-    expect(comp.mode).toBe(ViewerMode.PAGE);
+    expect(modeService.mode).toBe(ViewerMode.PAGE);
   }));
 
   it('should change to PAGE_ZOOMED-mode when doubleclicking in PAGE-mode', fakeAsync(() => {
```
Comment (raymondkarstensen) [inline src/lib/src/viewer/viewer.component.spec.ts]: Should this test be moved to ModeService?

File: src/lib/src/viewer/viewer.component.spec.ts [hunk 8]
```diff
@@ -256,42 +255,42 @@ describe('ViewerComponent', function () {
...
     const firstOverlay = viewerService.getOverlays()[0];
     const clickEvent = createClickEvent(firstOverlay);
     clickService.click(clickEvent);
     clickService.click(clickEvent);
     tick(1000);
-    expect(comp.mode).toBe(ViewerMode.PAGE_ZOOMED);
+    expect(modeService.mode).toBe(ViewerMode.PAGE_ZOOMED);
   }));
 
   it('should change to PAGE-mode when doubleclick in PAGE_ZOOMED-mode', fakeAsync(() => {
```
Comment (raymondkarstensen) [inline src/lib/src/viewer/viewer.component.spec.ts]: Should this test be moved to ModeService?

File: src/lib/src/viewer/viewer.component.spec.ts [hunk 9]
```diff
@@ -346,7 +345,7 @@ describe('ViewerComponent', function () {
 
 
   it('should change page when swipeing to left', () => {
```
Comment (raymondkarstensen) [inline src/lib/src/viewer/viewer.component.spec.ts]: Should this test be moved to ModeService?

Comment (ooystein) [reply src/lib/src/viewer/viewer.component.spec.ts]: I have now deleted these tests as they were quite similar to some e2e tests. Then I added some new unit tests for the ModeService.

**Pull request merged**